python/src/indices.rs

```diff
@@ -269,12 +287,97 @@ pub fn shuffle_transformed_vectors(
     }
 }
 
+async fn do_load_shuffled_vectors(
+    filenames: Vec<String>,
+    dir_path: &str,
+    dataset: &Dataset,
+    column: &str,
+    mut ivf_model: IvfModel,
+    pq_model: ProductQuantizer,
+) -> PyResult<()> {
+    let (_, path) = object_store_from_uri_or_path(dir_path).await?;
+    let streams = load_partitioned_shuffles(path.clone(), filenames)
+        .await
+        .infer_error()?;
+
+    let obj_store = dataset.ds.object_store();
+    let path = dataset.ds.indices_dir().child("shuffled_vectors.idx");
+    let mut writer = obj_store.create(&path).await.infer_error()?;
+    write_pq_partitions(&mut writer, &mut ivf_model, Some(streams), None)
+        .await
+        .infer_error()?;
+    let index_name = "ivf_pq_index";
+
+    let metadata = IvfPQIndexMetadata::new(
+        index_name.to_string(),
+        column.to_string(),
+        dataset.ds.version().version,
+        pq_model.distance_type,
+        ivf_model,
+        pq_model,
+        vec![],
+    );
+
+    let metadata = Index::try_from(&metadata).infer_error()?;
+    let pos = writer.write_protobuf(&metadata).await.infer_error()?;
+    writer.write_magics(pos, 0, 1, MAGIC).await.infer_error()?;
+    writer.shutdown().await.infer_error()?;
+
+    let mut ds = dataset.ds.as_ref().clone();
+    ds.commit_existing_index(index_name, column)
+        .await
+        .infer_error()?;
```

Comment:
A lot of this logic seems pretty odd to be in `python`.  Maybe we want to move most of this into a method in `rust/lance/src/index/vector/ivf.rs`.  This way we don't need to make `object_store` and `indices_dir` public.